File: runtime/flutter/painting/edgeInsets.ts

import { IEdgeInsetsGeometry } from "./edgeInsetsGeometry";
import { Axis } from "./axis";
import { ISize } from "../../dart/ui/size";
import { TextDirection } from "../../dart/ui/textDirection";
import { IOffset } from "../../dart/ui/offset";
import { IRect } from "../../dart/ui/rect";
declare const flutter: {
painting: {
edgeInsetsLerp: (a: IEdgeInsets | undefined, b: IEdgeInsets | undefined, t: number) => IEdgeInsets | undefined
}
};
export interface IEdgeInsets


{
left: number;
top: number;
right: number;
bottom: number;
getTopLeft: () => IOffset;
getTopRight: () => IOffset;
getBottomLeft: () => IOffset;
getBottomRight: () => IOffset;
getFlipped: () => IEdgeInsets;
inflateRect: (rect: IRect) => IRect;
deflateRect: (rect: IRect) => IRect;
subtract: (other: IEdgeInsetsGeometry) => IEdgeInsetsGeometry;
add: (other: IEdgeInsetsGeometry) => IEdgeInsetsGeometry;
clamp: (min: IEdgeInsetsGeometry, max: IEdgeInsetsGeometry) => IEdgeInsetsGeometry;
resolve: (direction?: TextDirection | undefined) => IEdgeInsets;
copyWith: ( props? : {  bottom? : number | undefined, left? : number | undefined, right? : number | undefined, top? : number | undefined,}) => IEdgeInsets;
along: (axis: Axis) => number;
inflateSize: (size: ISize) => ISize;
deflateSize: (size: ISize) => ISize;
toString: () => string;
getIsNonNegative: () => boolean;
getHorizontal: () => number;
getVertical: () => number;
getCollapsedSize: () => ISize;
getHashCode: () => number;
}export class EdgeInsets

 implements IEdgeInsetsGeometry
{    public static zero = EdgeInsets.only();
    public readonly left: number = undefined as any;
    public readonly top: number = undefined as any;
    public readonly right: number = undefined as any;
    public readonly bottom: number = undefined as any;
public static lerp(a: IEdgeInsets | undefined, b: IEdgeInsets | undefined, t: number) : IEdgeInsets | undefined{
return flutter.painting.edgeInsetsLerp(a, b, t);
}    private readonly _dart_getTopLeft: () => IOffset = undefined as any;
    private readonly _dart_getTopRight: () => IOffset = undefined as any;
    private readonly _dart_getBottomLeft: () => IOffset = undefined as any;
    private readonly _dart_getBottomRight: () => IOffset = undefined as any;
    private readonly _dart_getFlipped: () => IEdgeInsets = undefined as any;
    private readonly _dart_inflateRect: (rect: IRect) => IRect = undefined as any;
    private readonly _dart_deflateRect: (rect: IRect) => IRect = undefined as any;
    private readonly _dart_subtract: (other: IEdgeInsetsGeometry) => IEdgeInsetsGeometry = undefined as any;
    private readonly _dart_add: (other: IEdgeInsetsGeometry) => IEdgeInsetsGeometry = undefined as any;
    private readonly _dart_clamp: (min: IEdgeInsetsGeometry, max: IEdgeInsetsGeometry) => IEdgeInsetsGeometry = undefined as any;
    private readonly _dart_resolve: (direction?: TextDirection | undefined) => IEdgeInsets = undefined as any;
    private readonly _dart_copyWith: ( props? : {  bottom? : number | undefined, left? : number | undefined, right? : number | undefined, top? : number | undefined,}) => IEdgeInsets = undefined as any;
    private readonly _dart_along: (axis: Axis) => number = undefined as any;
    private readonly _dart_inflateSize: (size: ISize) => ISize = undefined as any;
    private readonly _dart_deflateSize: (size: ISize) => ISize = undefined as any;
    private readonly _dart_toString: () => string = undefined as any;
    private readonly _dart_getIsNonNegative: () => boolean = undefined as any;
    private readonly _dart_getHorizontal: () => number = undefined as any;
    private readonly _dart_getVertical: () => number = undefined as any;
    private readonly _dart_getCollapsedSize: () => ISize = undefined as any;
    private readonly _dart_getHashCode: () => number = undefined as any;
public getTopLeft() : IOffset {
    return this._dart_getTopLeft();
}
public getTopRight() : IOffset {
    return this._dart_getTopRight();
}
public getBottomLeft() : IOffset {
    return this._dart_getBottomLeft();
}
public getBottomRight() : IOffset {
    return this._dart_getBottomRight();
}
public getFlipped() : IEdgeInsets {
    return this._dart_getFlipped();
}
public inflateRect(rect: IRect) : IRect {
    return this._dart_inflateRect(rect);
}
public deflateRect(rect: IRect) : IRect {
    return this._dart_deflateRect(rect);
}
public subtract(other: IEdgeInsetsGeometry) : IEdgeInsetsGeometry {
    return this._dart_subtract(other);
}
public add(other: IEdgeInsetsGeometry) : IEdgeInsetsGeometry {
    return this._dart_add(other);
}
public clamp(min: IEdgeInsetsGeometry, max: IEdgeInsetsGeometry) : IEdgeInsetsGeometry {
    return this._dart_clamp(min, max);
}
public resolve(direction?: TextDirection | undefined) : IEdgeInsets {
    return this._dart_resolve(direction);
}
public copyWith( props? : {  bottom? : number | undefined, left? : number | undefined, right? : number | undefined, top? : number | undefined,}) : IEdgeInsets {
    return this._dart_copyWith(props);
}
public along(axis: Axis) : number {
    return this._dart_along(axis);
}
public inflateSize(size: ISize) : ISize {
    return this._dart_inflateSize(size);
}
public deflateSize(size: ISize) : ISize {
    return this._dart_deflateSize(size);
}
public toString() : string {
    return this._dart_toString();
}
public getIsNonNegative() : boolean {
    return this._dart_getIsNonNegative();
}
public getHorizontal() : number {
    return this._dart_getHorizontal();
}
public getVertical() : number {
    return this._dart_getVertical();
}
public getCollapsedSize() : ISize {
    return this._dart_getCollapsedSize();
}
public getHashCode() : number {
    return this._dart_getHashCode();
}
}

File: lib/cfr/builtins/libs/flutter/painting/edgeInsets.dart
import 'dart:core' as _fac9;
import 'dart:ui' as _a643;

import 'package:flutter/src/painting/basic_types.dart' as _487f;
import 'package:flutter/src/painting/edge_insets.dart' as _96d1;

import 'package:hydro_sdk/cfr/runtimeSupport.dart' as _36c2;

class VMManagedEdgeInsets extends _36c2.VMManagedBox<_96d1.EdgeInsets> {VMManagedEdgeInsets({required this.table, required this.vmObject, required this.hydroState}) : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) { table  [
'left'
] = vmObject.left;
table  [
'top'
] = vmObject.top;
table  [
'right'
] = vmObject.right;
table  [
'bottom'
] = vmObject.bottom;
table  [
'getTopLeft'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Offset>(object: vmObject.topLeft, hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'getTopRight'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Offset>(object: vmObject.topRight, hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'getBottomLeft'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Offset>(object: vmObject.bottomLeft, hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'getBottomRight'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Offset>(object: vmObject.bottomRight, hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'getFlipped'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_96d1.EdgeInsets>(object: vmObject.flipped, hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'inflateRect'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Rect>(object: vmObject.inflateRect(_36c2.maybeUnBoxAndBuildArgument<_a643.Rect, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'deflateRect'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Rect>(object: vmObject.deflateRect(_36c2.maybeUnBoxAndBuildArgument<_a643.Rect, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'subtract'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_96d1.EdgeInsetsGeometry>(object: vmObject.subtract(_36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsetsGeometry, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'add'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_96d1.EdgeInsetsGeometry>(object: vmObject.add(_36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsetsGeometry, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'clamp'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 
return [_36c2.maybeBoxObject<_96d1.EdgeInsetsGeometry>(object: vmObject.clamp(_36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsetsGeometry, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState), _36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsetsGeometry, _fac9.dynamic>(luaCallerArguments  [
2
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'resolve'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_96d1.EdgeInsets>(object: vmObject.resolve(_36c2.maybeUnBoxEnum(values: _a643.TextDirection.values, boxedEnum: luaCallerArguments  [
1
])), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'copyWith'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 


return [_36c2.maybeBoxObject<_96d1.EdgeInsets>(object: vmObject.copyWith(bottom: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'bottom'
] : null?.toDouble(), left: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'left'
] : null?.toDouble(), right: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'right'
] : null?.toDouble(), top: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'top'
] : null?.toDouble()), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'along'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.along(_36c2.maybeUnBoxEnum(values: _487f.Axis.values, boxedEnum: luaCallerArguments  [
1
])),]; } );
table  [
'inflateSize'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Size>(object: vmObject.inflateSize(_36c2.maybeUnBoxAndBuildArgument<_a643.Size, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'deflateSize'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Size>(object: vmObject.deflateSize(_36c2.maybeUnBoxAndBuildArgument<_a643.Size, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'toString'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.toString(),]; } );
table  [
'getIsNonNegative'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.isNonNegative,]; } );
table  [
'getHorizontal'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.horizontal,]; } );
table  [
'getVertical'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.vertical,]; } );
table  [
'getCollapsedSize'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Size>(object: vmObject.collapsedSize, hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'getHashCode'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.hashCode,]; } ); }

final _36c2.HydroTable table;

final _36c2.HydroState hydroState;

final _96d1.EdgeInsets vmObject;

 }

void loadEdgeInsets({required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { 
table  [
'edgeInsetsLerp'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 

final returnValue = _96d1.EdgeInsets.lerp(_36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsets?, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState), _36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsets?, _fac9.dynamic>(luaCallerArguments  [
2
], parentState: hydroState), luaCallerArguments  [
3
]?.toDouble());if(returnValue!= null){return [_36c2.maybeBoxObject<_96d1.EdgeInsets?>(object: returnValue, hydroState: hydroState, table: _36c2.HydroTable()),];}return []; } );
_36c2.registerBoxer<_96d1.EdgeInsets>(boxer: ({required _96d1.EdgeInsets vmObject, required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { return  VMManagedEdgeInsets(vmObject: vmObject, hydroState: hydroState, table: table); } ); } 
